test/puppetlabs/puppetdb/cli/services_test.clj

```diff
@@ -493,27 +497,38 @@
                             (let [result (f first?)]
                               (.await after-gc)
                               result))
-          log (atom [])]
-      (with-redefs [svcs/invoke-periodic-gc invoke-periodic]
+          log (atom [])
+          current-time (now)
+          six-days (.minus current-time (time/parse-period "6d"))
+          seven-days (.minus current-time (time/parse-period "7d"))
+          basic-date1 (time/unparse (time/formatters :basic-date) six-days)
+          basic-date2 (time/unparse (time/formatters :basic-date) seven-days)
+          iso_date1 (time/unparse (time/formatters :date-hour-minute-second) six-days)
+          iso_date2 (time/unparse (time/formatters :date-hour-minute-second) seven-days)
+          ago-mock (fn [_]
+                    (.minus current-time (time/parse-period "2d")))]
+      (with-redefs [svcs/invoke-periodic-gc invoke-periodic
+                    time/ago ago-mock]
```

Comment:
Why did we need to redef the `ago` func here? I tried running the test without doing so and the test hung and I had to manually kill the pids with `pg_cancel_backend` for the queries the test issued. The test was also still failing for me locally when I ran it without redefing `ago`. If trying to fix up this test is too much of a pain feel free to ignore it for now, I can help look at it more tomorrow. I think there are some tricky bits there, iirc

Reply:
The ago redef is needed in order to have the gc-bulldozer working properly because it uses both `report-ttl` and `resource-event-ttl` when cleaning up the old reports and resource event partitions. A time shift is simulated so reports are created and valid, but when gc is called, the resources are already expired and allowing it to cut the execution of the long running pids.

Reply:
I'm still going to need time to poke at this test. I'm wondering if changing the timestamps for the events in the reports that are submitted get us around needing to redef the `ago` call?